MapboxCoreNavigation/RouteController.swift

```diff
@@ -634,6 +645,7 @@ extension RouteController: CLLocationManagerDelegate {
         updateDistanceToIntersection(from: location)
         updateRouteStepProgress(for: location)
         updateRouteLegProgress(for: location)
+        detectRouteStepInTunnel(for: location)
```

Comment:
Based on the simplification suggested in https://github.com/mapbox/mapbox-navigation-ios/pull/1127#discussion_r169512237, I think we should actually inline this calculation, much as you had originally. That way:

* RouteStepProgress can keep track of the last intersection that the user has passed, either as an object or an index into the step’s `intersections` array.
* RouteControllerDelegate can react to `routeControllerProgressDidChange` or any other notification by checking whether the current step progress’s `lastIntersectionPassed` has `tunnel` in its `outletRoadClasses`.
* No new delegate methods would be required, and future features tied to freeway driving can piggyback on the same idea to check for the `motorway` class without requiring lots of new code.

Reply:
> RouteStepProgress can keep track of the last intersection that the user has passed, either as an object or an index into the step’s `intersections` array.

This would require something slightly different than what https://github.com/mapbox/mapbox-navigation-ios/pull/1127#discussion_r169512237 suggests: perhaps when advancing to a new step, we’d iterate over all the intersections, gathering their distances from the start of the step. That way we’d only need to perform a simple lookup on each location update.

Reply:
@1ec5 hey I came up with a rather simplified approach with detecting multiple tunnel intersections on a route step. Please refer to the `RouteStep` extension - `tunnelIntersectionsBounds` which I consume in the `hasEnteredTunnel(_: coordinates)` function. It handles our edge case.